### PR TITLE
javax to jakarta, groovy 4, spring framework 6 and spring boot 3 migrations

### PR DESCRIPTION
**Pull request by jamesfredley**
PR so that a 9.0.0-SNAPSHOT can be published and we can continue working on other grails dependencies while we work on the failing tests ignored on this PR.  https://github.com/grails/gorm-hibernate5/issues/879

Requires https://github.com/grails/grails-data-mapping/pull/1810 changes and org.grails:grails-datastore-*:9.0.0-SNAPSHOT

Key Changes and Issues:

Groovy-5106 issues:

`The interface GormEntity cannot be implemented more than once with different arguments: org.grails.datastore.gorm.GormEntity<grails.gorm.tests.XXX> and org.grails.datastore.gorm.GormEntity<grails.gorm.tests.XXX>`

- https://github.com/grails/gorm-hibernate5/pull/878/files#diff-c17e8e625456e19789dac95321b96d288b570a3869500d5b6c892ddb89c12acd

examples-grails-data-service

`General error during canonicalization: No signature of method: org.codehaus.groovy.ast.ClassCodeExpressionTransformer.transform() is applicable for argument types: (org.codehaus.groovy.ast.expr.MethodCallExpression) values: [org.codehaus.groovy.ast.expr.MethodCallExpression@776b49b2[object: org.codehaus.groovy.ast.expr.VariableExpression@30b9eadd[variable: response] method: org.codehaus.groovy.ast.expr.ConstantExpression@497570fb[status] arguments: org.codehaus.groovy.ast.expr.ArgumentListExpression@412c995d[org.codehaus.groovy.ast.expr.PropertyExpression@3249a1ce[object: org.codehaus.groovy.ast.expr.ClassExpression@4dd94a58[type: org.springframework.http.HttpStatus] property: org.codehaus.groovy.ast.expr.ConstantExpression@2f4919b0[UNPROCESSABLE_ENTITY]]]]]`

- https://github.com/grails/gorm-hibernate5/pull/878/files#diff-7f825392aa37acd1cee0c2e7b9bb7366ad6eac64f3e6cdd816e156bcb69d30de

https://github.com/grails/gorm-hibernate5/pull/878/files#diff-12879bcb16768df7b12cc0afc7feb7dec6daac16cc78ab20d97a86461a37771f
`java.lang.IllegalStateException: Either class [example.Customer] is not a domain class or GORM has not been initialized correctly or has already been shutdown. Ensure GORM is loaded and configured correctly before calling any methods on a GORM entity.`

https://github.com/grails/gorm-hibernate5/pull/878/files#diff-b959418ad6dfb5509464cf7b01fe22b84cf1d72f4756299d9382f2f5afded695
`java.lang.IllegalStateException: Either class [example.Book] is not a domain class or GORM has not been initialized correctly or has already been shutdown. Ensure GORM is loaded and configured correctly before calling any methods on a GORM entity.`
`java.lang.IllegalStateException: Either class [example.Book] is not a domain class or GORM has not been initialized correctly or has already been shutdown. Ensure GORM is loaded and configured correctly before calling any methods on a GORM entity.`

Additional currently failing tests.

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -10,10 +10,10 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        java: ['11']
+        java: ['17']
```
Comment (matrei) [inline .github/workflows/docs.yml]: Do we really need a matrix here?

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -10,10 +10,10 @@ jobs:
...
     env:
-      GIT_USER_NAME: puneetbehl
-      GIT_USER_EMAIL: behlp@unityfoundation.io
+      GIT_USER_NAME: 'grails-build'
+      GIT_USER_EMAIL: 'grails-build@users.noreply.github.com'
```
Comment (matrei) [inline .github/workflows/docs.yml:13]: These could be used at the bottom in the "Publish to Github Pages" step.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -7,10 +7,10 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        java: ['11']
+        java: ['17']
```
Comment (matrei) [inline .github/workflows/release.yml]: Do we need a matrix here?

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -7,10 +7,10 @@ jobs:
...
     env:
-      GIT_USER_NAME: puneetbehl
-      GIT_USER_EMAIL: behlp@unityfoundation.io
+      GIT_USER_NAME: 'grails-build'
+      GIT_USER_EMAIL: 'grails-build@users.noreply.github.com'
```
Comment (matrei) [inline .github/workflows/release.yml:10]: We can use these instead of Puneets details below.

File: .github/workflows/docs.yml (hunk 1 above)
Comment (jamesfredley) [reply .github/workflows/docs.yml]: I've made all of these github workflow changes.

**Comment by jamesfredley** [comment]
The failing tests on these changes will require a material effort to resolve, but for now should wait until the Grails 7 apps start.

**Review by matrei — COMMENTED** [review]
You know, I don't think the joint workflow is working as intended. It builds a Groovy Snapshot, but then fetches the published Snapshot from the Snapshot repo.

I think we should do `publishToMavenLocal` instead of `install` from the groovy build. Then the grails build will find the fresh Snapshot by having a `mavenLocal()` repo.

**Comment by jamesfredley** [comment]
I have updated this repo to use `publishToMavenLocal`.

**Review by matrei — APPROVED** [review]

**Pull request merged**